cpp/src/backends/protocol/otf_message.cc

```diff
@@ -1,30 +1,41 @@
-#include <arpa/inet.h>
-#include <sys/socket.h>
-#include <glog/logging.h>
-
 #include "src/backends/protocol/otf_message.hh"
 
 namespace torchserve {
-  byte_buffer OTFMessage::CreateLoadModelResponse(StatusCode code, const std::string& message) {
-    LoadModelResponse response = {
-      code, 
-      static_cast<int>(message.length()), 
-      message
-    };
-    std::byte msg[sizeof(LoadModelResponse)];
-    std::memcpy(msg, &response, sizeof(LoadModelResponse));
-    byte_buffer response_byte_buffer;
-    std::copy(response_byte_buffer.begin(), response_byte_buffer.end(), msg);
-    return response_byte_buffer;
+  void OTFMessage::EncodeLoadModelResponse(std::unique_ptr<torchserve::LoadModelResponse> response, char* data) {
+    char* p = data;
+    // Serialize response code
+    int32_t s_code = htonl(response->code);
+    memcpy(p, &s_code, sizeof(s_code));
+    p += sizeof(s_code);
+    // Serialize response message length
+    int32_t resp_length = htonl(response->length);
+    memcpy(p, &resp_length, sizeof(resp_length));
+    p += sizeof(resp_length);
+    // Serialize response message
+    strcpy(p, response->buf.c_str());
+    p += response->length;
+    // Expectation from frontend deserializer is a -1
+    // at the end of a LoadModelResponse
+    int32_t no_predict = htonl(response->predictions);
+    memcpy(p, &no_predict, sizeof(no_predict));
+    p += sizeof(no_predict);
   }
 
-  char OTFMessage::RetrieveCmd(Socket conn) {
+  bool OTFMessage::SendLoadModelResponse(const ISocket& client_socket_, std::unique_ptr<torchserve::LoadModelResponse> response) {
+    char *data = new char[sizeof(LoadModelResponse)];
```

Comment:
does the `sizeof` operator on `LoadModelResponse` object always yield the right size of the data? In the case of inference response, it did not return the expected response probably because it holds pointers to the data instead of the actual data.

Reply:
Will fix this